### PR TITLE
RavenDB-6178 multi-get now handles caching correctly

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.NewClient/Commands/MultiGetCommand.cs [hunk 1]
```diff
@@ -14,66 +15,189 @@ namespace Raven.NewClient.Client.Commands
...
+                    if (result.Modifications == null)
+                        result.Modifications = new DynamicJsonValue(result);
+
+                    result.Modifications[nameof(Response.Result)] = cachedResponse;
+                    anyModifications = true;
+                }
+            }
+
+            if (anyModifications)
+                response = _context.ReadObject(response, "multi_get/response");
```
Comment (ayende) [inline src/Raven.NewClient/Commands/MultiGetCommand.cs:147]: Why are you doing this?

File: src/Raven.NewClient/Commands/MultiGetCommand.cs [hunk 2]
```diff
@@ -14,66 +15,189 @@ namespace Raven.NewClient.Client.Commands
...
 
             Result = JsonDeserializationClient.BlittableArrayResult(response);
         }
 
+        protected override void CacheResponse(HttpCache cache, RequestExecuterOptions options, string url, HttpResponseMessage response, BlittableJsonReaderObject responseJson)
```
Comment (ayende) [inline src/Raven.NewClient/Commands/MultiGetCommand.cs:152]: You read it all as a single blittable, and that is wrong. What you want to do instead is to break into independent values _as you parse it_, like we do in smuggerl, or @aviviadi work on the BatchHandler.

File: src/Raven.NewClient/Commands/MultiGetCommand.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.NewClient/Commands/MultiGetCommand.cs:147]: JsonDeserializationClient is not supporting modifications?

File: src/Raven.NewClient/Commands/MultiGetCommand.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.NewClient/Commands/MultiGetCommand.cs:152]: Not possible here I think?

Comment (ppekrol) [reply src/Raven.NewClient/Commands/MultiGetCommand.cs:152]: I think that I can override ProcessResponse...

File: src/Raven.NewClient/Commands/MultiGetCommand.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.NewClient/Commands/MultiGetCommand.cs:147]: By design, but you shouldn't deal with it directly anyway here. build the blittable directly

**Pull request merged**